crates/core_arch/src/arm_shared/neon/mod.rs

```diff
@@ -3704,6 +3704,16 @@ pub unsafe fn vdupq_n_f32(value: f32) -> float32x4_t {
     float32x4_t(value, value, value, value)
 }
 
+/// Duplicate vector element to vector or scalar
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "fp-armv8,v8"))]
+#[cfg_attr(all(test, target_arch = "arm"), assert_instr("vdup.32"))]
+#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr(dup))]
+unsafe fn vdupq_n_f32_v8(value: f32) -> float32x4_t {
```

Comment:
Can you add a comment explaining why this is needed? Also this should probably be under `#[cfg(target_arch = "arm")]`.

Reply:
@Amanieu Will do, but first I will fix arm fused multiply-add to only require vfp4 instead of v8 (PR forthcoming) and then the fn name will change to `vdupq_n_f32_vfp4`.